### PR TITLE
[BugFix] add thrift max_message_size configuration

### PR DESCRIPTION
**Pull request by packy92**
## Why I'm doing:
When upgrade thrift to 0.20.0, the MaxMessageSize member defines the maximum size of a (received) message, in bytes.
The default value is represented by a constant named DEFAULT_MAX_MESSAGE_SIZE, whose value is 100 * 1024 * 1024 bytes.
This will cause FE to fail during deserialization when the returned result set is larger than 100M. Therefore, we add thrift max_message_size configuration and set a default value of 1G and the maximum configurable value is 2G.

## What I'm doing:
Add FE/BE configuration to increase MaxMessageSize when necessary.
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

## Documentation PRs only:

If you are submitting a PR that adds or changes English documentation and have not
included Chinese documentation, then you can check the box to request GPT to translate the
English doc to Chinese. Please ensure to uncheck the **Do not translate** box if translation is needed.
The workflow will generate a new PR with the Chinese translation after this PR is merged.

- [ ] Yes, translate English markdown files with GPT
- [x] Do not translate

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -1396,4 +1396,10 @@ CONF_mString(try_release_resource_before_core_dump, "data_cache");
 // Experimental feature, this configuration will be removed after testing is complete.
 CONF_mBool(lake_enable_alter_struct, "true");
 
+CONF_mInt32(thrift_max_message_size, "1073741824"); // 1 GB
```
Comment (kevincai) [inline be/src/common/config.h:1399]: better add some comments explaining a little bit about the parameters and possibly under what condition, may consider tune these parameters.

File: fe/fe-core/src/main/java/com/starrocks/rpc/ConfigurableTProtocolFactory.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import org.apache.thrift.protocol.TProtocolFactory;
+import org.apache.thrift.protocol.TSimpleJSONProtocol;
+import org.apache.thrift.transport.TTransport;
+
+public class ConfigurableTProtocolFactory {
+
+    public static TProtocolFactory getTProtocolFactory(String protocol) {
+        TProtocolFactory factory;
+        TConfiguration configuration = buildTConfiguration();
+        if (StringUtils.equalsIgnoreCase(protocol, "compact")) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/rpc/ConfigurableTProtocolFactory.java:33]: possibly make the protocol as enum type?

File: test/sql/test_conf/T/test_configure [hunk 3]
```diff
@@ -0,0 +1,32 @@
+-- name: test_configure
+admin set frontend config ("thrift_max_message_size"="10485760");
```
Comment (kevincai) [inline test/sql/test_conf/T/test_configure:2]: shall reset these configurations after test done, and mark the test case as @sequential, so it won't break other cases running in parallel.

The same to the be_configs.

**Review by kevincai — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/rpc/AttachmentRequest.java [hunk 4]
```diff
@@ -39,16 +35,7 @@ public class AttachmentRequest {
...
-        if (StringUtils.equalsIgnoreCase(protocol, "compact")) {
-            serializer = new TSerializer(TCompactProtocol::new);
-        } else if (StringUtils.equalsIgnoreCase(protocol, "json")) {
-            serializer = new TSerializer(TJSONProtocol::new);
-        } else {
-            // default bianry
-            serializer = new TSerializer(TBinaryProtocol::new);
-        }
-        return serializer;
+        return ConfigurableSerDesFactory.getTSerializer(protocol);
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/rpc/AttachmentRequest.java:38]: better to check the protocol value?

File: fe/fe-core/src/main/java/com/starrocks/rpc/ConfigurableSerDesFactory.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import org.apache.thrift.transport.TTransportException;
+
+public class ConfigurableSerDesFactory {
+
+    public static TSerializer getTSerializer() throws TTransportException {
+        return getTSerializer(Protocol.BINARY.name());
+    }
+
+    public static TSerializer getTSerializer(String protocol) throws TTransportException {
+        Protocol p = Protocol.getProtocol(protocol);
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/rpc/ConfigurableSerDesFactory.java:29]: better to check the protocol value?

**Review by Seaven — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**